CMakeLists.txt

```diff
@@ -424,26 +427,20 @@ else()
   target_compile_definitions(preciceCore PUBLIC PRECICE_NO_PETSC)
 endif()
 
+add_library(preciceExternal INTERFACE)
+
 # Setup Ginkgo
 if (PRECICE_FEATURE_GINKGO_MAPPING)
   # Next we configure the common kernels for the matrix assembly
-  # This CMake part is still a shortcoming from the experimental Ginkgo we are using right now
-  # TODO: Remove as soon as GPU feature is in main branch
-  set(Ginkgo_Path "${Ginkgo_DIR}/.." CACHE STRING "Path to the experimental Ginkgo Repository of branch a195f856e.")
-  include(${Ginkgo_Path}/cmake/unified_kernels.cmake)
-
-  # This command registers Ginkgo kernels for different execution backends
-  ginkgo_add_unified_kernels(preciceCore src/mapping/device/GinkgoRBFKernels.cpp)
-  # shortcoming of the kernel registration: we need to add the includes here to all generated targets
-  include_directories(${PROJECT_SOURCE_DIR}/src)
-  include_directories(${EIGEN3_INCLUDE_DIR} ${PROJECT_SOURCE_DIR}/thirdparty/fmt/include)
-  target_compile_definitions(preciceCore_cuda PUBLIC EIGEN_NO_CUDA)
-  target_compile_definitions(preciceCore_hip PUBLIC EIGEN_NO_CUDA)
-  set_target_properties(preciceCore_cuda PROPERTIES POSITION_INDEPENDENT_CODE ON)
-  set_target_properties(preciceCore_reference PROPERTIES POSITION_INDEPENDENT_CODE ON)
-  set_target_properties(preciceCore_hip PROPERTIES POSITION_INDEPENDENT_CODE ON)
-  set_target_properties(preciceCore_omp PROPERTIES POSITION_INDEPENDENT_CODE ON)
-  set_target_properties(preciceCore_dpcpp PROPERTIES POSITION_INDEPENDENT_CODE ON)
+  add_library(preciceCore_ginkgo STATIC
+          src/mapping/device/GinkgoRBFKernels.cpp
+          src/mapping/device/Ginkgo.cpp)
+  target_link_libraries(preciceCore_ginkgo PRIVATE Ginkgo::ginkgo Kokkos::kokkos preciceCore)
+  target_compile_definitions(preciceCore_ginkgo PUBLIC EIGEN_NO_CUDA)
```

Comment:
Eigen 'only' provides templates, which are implicitly inline, so I don't see that this violates the ODR rule.

Reply:
Templates reduce to symbols, which still clash if their implementation differs. This also holds for the solver btw, which needs to be compiled with the same macros. It's essentially the same reason Jaust ran into strange behaviour due to conflicting destructors in Eigen.

Reply:
I don't really understand how inline code can clash, but we can just take over the compile definition to preciceCore. I would assume that this doesn't make a difference anyhow, since the definition makes the functions then mostly callable and executable on a device, which is not what we compile anywhere.